src/zkp/pilog.rs

```diff
@@ -330,29 +339,56 @@ mod tests {
         paillier::DecryptionKey,
         ring_pedersen::VerifiedRingPedersen,
         utils::{random_plusminus_by_size_with_minimum, testing::init_testing},
+        zkp::BadContext,
     };
 
-    fn random_paillier_log_proof<R: RngCore + CryptoRng>(rng: &mut R, x: &BigNumber) -> Result<()> {
-        let (decryption_key, _, _) = DecryptionKey::new(rng)?;
+    fn random_paillier_log_proof<R: RngCore + CryptoRng>(
+        rng: &mut R,
+        x: &BigNumber,
+    ) -> (PiLogProof, CommonInput, Transcript) {
```

Comment:
suggestion: This could return `Result<(PiLogProof, ... )>` so we can use the `?` notation instead of manually unwrapping.